cvat-core/src/object-utils.ts

```diff
@@ -356,3 +357,35 @@ export function rle2Mask(rle: number[], width: number, height: number): number[]
 
     return decoded;
 }
+
+export function propagateShapes(shapes: SerializedShape[], from: number, to: number): SerializedShape[] {
+    const getCopyFromShape = (shape: SerializedShape): SerializedShape => ({
```

Comment:
Probably we want somehow to re-use the same code in action and in `propagateObjectAsync`

Reply:
Some utility functions like `mask2Rle`, `rle2Mask` are already is a part of cvat-core external API

Reply:
Added this function to cvat-core api.